### DISCUSSION TITLE
How to PR if a plugin uses FFI

### DISCUSSION THREAD

**Discussion by zhixiongdu027** [apache/apisix#6433] [General]
As the title,

If some upstream services use private protocols, 
proxy plugin need to use ffi for protocol conversion.

**Comment by tzssangglass** [comment]
Just like using FFI in OpenResty: https://moonbingbing.gitbooks.io/openresty-best-practices/content/lua/FFI.html

**Comment by zhixiongdu027** [comment]
Sorry, I means  is:

> If I develop a plugin that uses FFI , How can I Pr this plugin to  apisix.
>  seem like apisix doesn't contain any c code.

And I had got answer from @membphis :
1.  create a lua rock
2.  push to luarocks community
3.  reference it as a dependency

Many TKS.
